web/src/lib/utils.ts

```diff
@@ -31,3 +33,91 @@ export const updateParamList = async ({ param, value, add }: UpdateParamAction)
 
   await goto(`?${searchParams.toString()}`, { replaceState: true, noScroll: true, keepFocus: true });
 };
+
+export const getJobName = (jobName: JobName) => {
+  const names: Record<JobName, string> = {
+    [JobName.ThumbnailGeneration]: 'Generate Thumbnails',
+    [JobName.MetadataExtraction]: 'Extract Metadata',
+    [JobName.Sidecar]: 'Sidecar Metadata',
+    [JobName.SmartSearch]: 'Smart Search',
+    [JobName.FaceDetection]: 'Face Detection',
+    [JobName.FacialRecognition]: 'Facial Recognition',
+    [JobName.VideoConversion]: 'Transcode Videos',
+    [JobName.StorageTemplateMigration]: 'Storage Template Migration',
+    [JobName.Migration]: 'Migration',
+    [JobName.BackgroundTask]: 'Background Tasks',
+    [JobName.Search]: 'Search',
+    [JobName.Library]: 'Library',
+  };
+
+  return names[jobName];
+};
+
+let _key: string | undefined;
```

Comment:
This is fine for now, but having a single global key wouldn't work in the case that we had a cloud version of immich serving multiple users and doing ssr. It's not a change from the current code and not a blocker, but I thought I'd mention it anyway. Could potentially be worth a TODO. The typical way to avoid such a situation is with context: https://github.com/sveltejs/kit/discussions/4339#discussioncomment-2372710